File: user/src/main/java/com/sparta/user/infrastructure/service/RedisService.java
package com.sparta.user.infrastructure.service;

import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Service;

import java.time.Duration;

@Service
@RequiredArgsConstructor
public class RedisService {

    private final RedisTemplate<String, Object> redisTemplate;

    public void set(String key, String value, Duration expiration) {
        redisTemplate.opsForValue().set(key, value, expiration);
    }

    public boolean hasKey(String key) {
        return redisTemplate.hasKey(key);
    }
}

File: hub/src/main/kotlin/com/sparta/hub/application/redis/RedisService.kt
package com.sparta.hub.application.redis

import com.sparta.hub.application.dto.RouteResult
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Service

@Service
class RedisService(
    private val redisTemplate: RedisTemplate<String, Any>
) {

    fun setHubRouteResult(key1: String, key2: String, value: RouteResult) {
        redisTemplate.opsForHash<String, RouteResult>().put(
            key1,
            key2,
            value
        )
    }

    fun getHubRouteResult(key1: String, key2: String): RouteResult? {
        return redisTemplate.opsForHash<String, RouteResult>().get(key1, key2)
    }

    fun getHubRouteByStartingWithKey(key: String): List<RouteResult> {
        return redisTemplate.opsForHash<String, RouteResult>().values(key)
    }


}